pkg/utils/has/components.go

```diff
@@ -135,7 +135,7 @@ func (h *HasController) WaitForComponentPipelineToBeFinished(component *appservi
 }
 
 // Universal method to create a component in the kubernetes clusters.
-func (h *HasController) CreateComponent(componentSpec appservice.ComponentSpec, namespace string, outputContainerImage string, secret string, applicationName string, skipInitialChecks bool, annotations map[string]string) (*appservice.Component, error) {
+func (h *HasController) CreateComponent(componentSpec appservice.ComponentSpec, namespace string, outputContainerImage string, secret string, applicationName string, skipInitialChecks bool, annotations map[string]string, shouldGenerateImage bool) (*appservice.Component, error) {
```

Comment:
Do we need to pass this `shouldGenerateImage` parameter? since for generating image we set annotations in the component CR, isn't it possible through just passing the value `constants.ImageControllerAnnotationRequestPublicRepo` for annotations in `CreateComponent`, even if we don't pass it? Don't get why do we need to change it here.

Reply:
[Component annotation](https://github.com/redhat-appstudio/image-controller#legacy-deprecated-component-image-repository) (`image.redhat.com/generate: '{"visibility": "public"}'`) will be deprecated soon.
In the newer approach, [ImageRepository CR](https://github.com/redhat-appstudio/image-controller#appstudio-component-image-repository), this annotation should not be used. That's why it was added `shouldGenerateImage`.

In the file [tests/remotesecret/image-repository-cr-image-pull-remote-secret.go](https://github.com/redhat-appstudio/e2e-tests/pull/814/files#diff-618e37058f5088e0bf346c0b2a74ee47f36105f25546ba89762888a12bbed9b1R101), `shouldGenerateImage` is false, so `constants.ImageControllerAnnotationRequestPublicRepo` will be not added.

Reply:
Oh, I see